Adding a blogpost for Stateful Functions 2.0 release announcement. (#321)
apache/flink-web: _posts/2020-04-02-release-statefun-2.0.0.md

</center>

<div style="line-height:150%;">
    <br>
</div>

Following the database analogy, embedded functions are a bit like _stored procedures_, but in a principled way: the functions here are normal Java/Scala/Kotlin functions implementing standard interfaces and can be developed or tested in any IDE.

## Loading Data into the Database

When building a new stateful application, you usually don’t start from a completely blank slate. Often, the application has initial state, such as initial “bootstrap” state, or state from previous versions of the application. When using a database, one could simply bulk load the data to prepare the application.

The equivalent step for Flink would be to write a [savepoint]({{ site.docs-stable }}/ops/state/savepoints.html) that contains the initial state. Savepoints are snapshots of the state of the distributed stream processing application and can be passed to Flink to start processing from that state. Think of them as a database dump, but of a distributed streaming database. In the case of StateFun, the savepoint would contain the state of the functions.

> To create a savepoint for a Stateful Functions program, check out the [State Bootstrapping API](https://ci.apache.org/projects/flink/flink-statefun-docs-master/deployment-and-operations/state-bootstrap.html) that is part of StateFun 2.0. The State Bootstrapping API uses Flink’s [DataSet API]({{ site.docs-stable }}/dev/batch/), but we plan to expand this to use SQL in the next versions.

Reviewer: ```suggestion
To create a savepoint for a Stateful Functions program, check out the [State Bootstrapping API](https://ci.apache.org/projects/flink/flink-statefun-docs-release-2.0/deployment-and-operations/state-bootstrap.html) that is part of StateFun 2.0. The State Bootstrapping API uses Flink’s [DataSet API]({{ site.docs-stable }}/dev/batch/), but we plan to expand this to use SQL in the next versions.
```

## Try it out and get involved!

> We hope that we could convey some of the excitement we feel about Stateful Functions. If we managed to pique your curiosity, try it out — for example, starting with [this walkthrough](https://ci.apache.org/projects/flink/flink-statefun-docs-master/getting_started/walkthrough.html).

Reviewer: Broken link:
```suggestion
We hope that we could convey some of the excitement we feel about Stateful Functions. If we managed to pique your curiosity, try it out — for example, starting with [this walkthrough](https://ci.apache.org/projects/flink/flink-statefun-docs-release-2.0/getting-started/python_walkthrough.html).
```

The project is still in a comparatively early stage, so if you want to get involved, there is lots to work on: SDKs for other languages (e.g. Go, JavaScript, Rust), ingresses/egresses and tools for testing, among others.

To follow the project and learn more, please check out these resources:

* Code: [https://github.com/apache/flink-statefun](https://github.com/apache/flink-statefun)

> * Stateful Functions Website: [https://statefun.io](https://statefun.io)

Reviewer: Is it a bit strange calling this "website", since this now redirects to the "What is Stateful Functions?" page?

> * Docs [https://ci.apache.org/projects/flink/flink-statefun-docs-master/](https://ci.apache.org/projects/flink/flink-statefun-docs-master/)

Reviewer: ```suggestion
* Docs [https://ci.apache.org/projects/flink/flink-statefun-docs-release-2.0/](https://ci.apache.org/projects/flink/flink-statefun-docs-release-2.0/)
```
